### ISSUE TITLE
Apparently mysql-server=5.7* wasn't found

### ISSUE THREAD

**Issue by hfabio** [arm32v7/mysql#1]
log error:
```
Fetched 13.7 MB in 1min 15s (181 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
E: Version '5.7*' for 'mysql-server' was not found
The command '/bin/sh -c { 		echo mysql-server mysql-server/data-dir select ''; 		echo mysql-server mysql-server/root-pass password ''; 		echo mysql-server mysql-server/re-root-pass password ''; 		echo mysql-server mysql-server/remove-test-db select false; 	} | debconf-set-selections 	&& apt-get update && apt-get install -y mysql-server="${MYSQL_VERSION}"* && rm -rf /var/lib/apt/lists/* 	&& rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql && chown -R mysql:mysql /var/lib/mysql' returned a non-zero code: 100
```

I bet its a problem at:
```
 apt-get install -y mysql-server="${MYSQL_VERSION}"* && rm -rf /var/lib/apt/lists/*
```

Can you give me a hand? I need a mysql 5.7 at my raspberry pi 4 buster

**Comment by kevlee** [comment]
Hello,

I had the sames issue. 

you could use yobasystems/alpine-mariadb images already compile for arm32v7

Its a fork of mysql so you didn't have to change the docker-compose or you connexion implementation.

Hope that will help you ;)

**Comment by hfabio** [comment]
> Hello,
> 
> I had the sames issue.
> 
> you could use yobasystems/alpine-mariadb images already compile for arm32v7
> 
> Its a fork of mysql so you didn't have to change the docker-compose or you connexion implementation.
> 
> Hope that will help you ;)

Thanks for your answer, but actually I really have to replicate my development environment, and for this reason I needed the same exact mysql version :(

Appreciate your tip, thanks!

**Comment by kevlee** [comment]
you can try to add before line 14 RUN deb http://http.us.debian.org/debian unstable main contrib non-free in your dockerfile.

The package mysql-server-5.7 is in sid repo cf:
https://packages.debian.org/**sid**/mysql-server-5.7
https://wiki.debian.org/fr/DebianRepository

**Comment by hfabio** [comment]
@kevlee 
I added the line below before line 14:
`RUN "deb http://http.us.debian.org/debian unstable main contrib non-free" | tee -a /etc/apt/sources.list.d/mysql.list`
apparently it works (at least didn't break the execution yet), but with a warning notification that the public GPG key wasn't found.
But now I'm getting this issue:
`Processing triggers for libc-bin (2.19-18+deb8u10) ...
Errors were encountered while processing:
 /var/cache/apt/archives/libcrypt1_1%3a4.4.17-1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
The command '/bin/sh -c apt-get update && apt-get install -y --force-yes perl --no-install-recommends && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100`
By the way thank you so much for your attention, appreciated a lot.

**Comment by kevlee** [comment]
Hello,

You are welcome. 

Concerning your issue you could try to ask a buster images:
"FROM resin/rpi-raspbian:buster"
instead of 
"FROM resin/rpi-raspbian"

Other troubleshooting that you could operate is:

- Removing the cache images with docker images rm before relaunch your compilation
- Have you compile the project with docker buildx?

I didn't try to compile that image because all thread tell: use the mariadb that is equivalent to mysql. 

You don't need to perform any changes in your implementation if you use mariadb. 

BR,

**Comment by hfabio** [comment]
I did build with `docker build -t mysql5.7:1.0 .`
No success yet :(
And yeah, I know mariadb is pretty similar/equivalent to mysql, but I really wanted to replicate all environment as they are to avoid some issues (for example, We use mysql5.7 in production, if mariadb did a fix in something that mysql didn't, we will have a silent issue, right?), and change the production database now isn't possible at the moment.

**Comment by kevlee** [comment]
You did docker build -t mysql5.7:1.0 on raspberry pi ?

BR,

**Comment by hfabio** [comment]
yeah, built from local Dockerfile (there is a dot at the end specifying the path to current directory) over SSH on my raspberry pi 4.

**Comment by kevlee** [comment]
try this docker file
[Dockerfile.txt](https://github.com/arm32v7/mysql/files/5861258/Dockerfile.txt)



it works for me. if you need i could share you a image.

BR

**Comment by hfabio** [comment]
I think it added the mysql repository pretty fine:
`
Get:10 http://deb.debian.org/debian unstable/main armhf mysql-common all 5.8+1.0.6 [7360 B]
Get:15 http://deb.debian.org/debian unstable/main armhf mysql-client-core-5.7 armhf 5.7.26-1+b1 [6383 kB]
Get:16 http://deb.debian.org/debian unstable/main armhf mysql-client-5.7 armhf 5.7.26-1+b1 [2163 kB]
Get:30 http://deb.debian.org/debian unstable/main armhf mysql-server-5.7 armhf 5.7.26-1+b1 [3020 kB]
Get:48 http://deb.debian.org/debian unstable/main armhf mysql-server all 5.7.26-1 [80.6 kB]
`
But I got this error:
`Unpacking libc6:armhf (2.31-9) over (2.28-10) ...
tar: ./control: Cannot utime: Operation not permitted
tar: ./md5sums: Cannot utime: Operation not permitted
tar: ./shlibs: Cannot utime: Operation not permitted
tar: ./symbols: Cannot utime: Operation not permitted
tar: ./triggers: Cannot utime: Operation not permitted
tar: .: Cannot utime: Operation not permitted
tar: Exiting with failure status due to previous errors
dpkg-deb: error: tar subprocess returned error exit status 2
dpkg: error processing archive /var/cache/apt/archives/libcrypt1_1%3a4.4.17-1_armhf.deb (--unpack):
 dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
 /var/cache/apt/archives/libcrypt1_1%3a4.4.17-1_armhf.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
The command '/bin/sh -c {               echo mysql-server mysql-server/data-dir select '';              echo mysql-server mysql-server/root-pass password '';          echo mysql-server mysql-server/re-root-pass password '';                echo mysql-server mysql-server/remove-test-db select false;    } | debconf-set-selections      && apt-get install -y mysql-server="${MYSQL_VERSION}"* && rm -rf /var/lib/apt/lists/*   && rm -rf /var/lib/mysql && mkdir -p /var/lib/mysql && chown -R mysql:mysql /var/lib/mysql' returned a non-zero code: 100`

It would be very helpful if you could share the image

**Comment by kevlee** [comment]
You can find it on my repos: https://hub.docker.com/repository/docker/kevlee/mysql5.7-rpi.

I did try to run but docker build didn't trow any issues.

BR

**Comment by hfabio** [comment]
I had `standard_init_linux.go:211: exec user process caused "no such file or directory"`
But I did realize your script starts with `ADD file:216371f6fe8c803a0191f84e555c809a3301d1344c62b831dddb5e0c595fe0ef in / `

**Comment by kevlee** [comment]
Hello,

I think the issue come from the entrypoint.sh. 

I couldn't take more time on it i try to release my personal project.

What you can do is to create a basic image with docker run -ti debian:buster bash and apply each line of the dockerfile. once it done you Copy the docker-entrypoint.sh and launch it inside the docker.

BR,
